SecondMax.java

```diff
@@ -0,0 +1,38 @@
+public class SecondMax {
+    static void main(String[] args) {
+        test(new int[]{1,2,3,4,5,6,7});
+        test(new int[]{});
+        test(new int[]{1});
+        test(new int[]{9,9,9});
+    }
```

Comment:
There should be test cases which cover other cases. For example:
1. Not sorted input.
2. Negative numbers
3. When second maximum is first, second, last or second last number in the array (to catch off by one bugs).